### PR TITLE
[Reward] - update reward endpoint part 2

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.monday.com/boards/2218344956/pulses/2852587684

### CODE REVIEW

**Review by soulBit — APPROVED** [review]

File: src/app/pages/RewardPage/components/RewardsDetail/index.tsx [hunk 1]
```diff
@@ -84,10 +83,10 @@ export const RewardsDetail: React.FC<IRewardsDetailProps> = ({
           <LoadableValue
             value={
               bignumber(availableAmount).greaterThan(0) ? (
-                <Tooltip content={`${weiTo18(availableAmount)} ${asset}`}>
+                <Tooltip content={`${availableAmount} ${asset}`}>
                   <>
                     {showApproximateSign && '≈ '}
-                    {weiToNumberFormat(availableAmount, 6)}
+                    {toNumberFormat(availableAmount, 6)}
                     <span className="tw-mr-1">...</span> {asset}
```
Comment (tiltom) [inline src/app/pages/RewardPage/components/RewardsDetail/index.tsx:90]: Use `AssetValue` component.

File: src/app/pages/RewardPage/components/RewardsDetail/index.tsx [hunk 2]
```diff
@@ -106,10 +105,10 @@ export const RewardsDetail: React.FC<IRewardsDetailProps> = ({
         <LoadableValue
           value={
             bignumber(totalEarnedAmount).greaterThan(0) ? (
-              <Tooltip content={`${weiTo18(totalEarnedAmount)} ${asset}`}>
+              <Tooltip content={`${totalEarnedAmount} ${asset}`}>
                 <>
                   {showApproximateSign && '≈ '}
-                  {weiToNumberFormat(totalEarnedAmount, 6)}
+                  {toNumberFormat(totalEarnedAmount, 6)}
                   <span className="tw-mr-1">...</span> {asset}
```
Comment (tiltom) [inline src/app/pages/RewardPage/components/RewardsDetail/index.tsx:112]: Use `AssetValue` component.

**Review by tiltom — CHANGES_REQUESTED** [review]

**Comment by tiltom** [comment]
Testnet inconsistencies, @BetsyBraddock and @pietro-maximoff , pls take a look at it.

New implementation:
image

Existing implementation:
image

**Review by Alitasovryn — DISMISSED** [review]

File: src/app/pages/RewardPage/components/RewardsDetail/index.tsx [hunk 3]
```diff
@@ -84,11 +84,16 @@ export const RewardsDetail: React.FC<IRewardsDetailProps> = ({
...
                     {showApproximateSign && '≈ '}
-                    {weiToNumberFormat(availableAmount, 6)}
-                    <span className="tw-mr-1">...</span> {asset}
+                    <AssetValue
+                      value={Number(availableAmount)}
+                      minDecimals={6}
+                      maxDecimals={6}
+                      assetString={asset}
+                      mode={AssetValueMode.auto}
+                    />
```
Comment (tiltom) [inline src/app/pages/RewardPage/components/RewardsDetail/index.tsx:96]:
```suggestion
                    <AssetValue
                      value={Number(availableAmount)}
                      minDecimals={6}
                      maxDecimals={6}
                      assetString={asset}
                      mode={AssetValueMode.auto}
                      isApproximation={showApproximateSign}
                    />
```

File: src/app/pages/RewardPage/components/RewardsDetail/index.tsx [hunk 4]
```diff
@@ -84,11 +84,16 @@ export const RewardsDetail: React.FC<IRewardsDetailProps> = ({
           <LoadableValue
             value={
               bignumber(availableAmount).greaterThan(0) ? (
-                <Tooltip content={`${weiTo18(availableAmount)} ${asset}`}>
+                <Tooltip content={`${availableAmount} ${asset}`}>
                   <>
                     {showApproximateSign && '≈ '}
```
Comment (tiltom) [inline src/app/pages/RewardPage/components/RewardsDetail/index.tsx:89]:
```suggestion

```

File: src/app/pages/RewardPage/components/RewardsDetail/index.tsx [hunk 5]
```diff
@@ -106,11 +111,16 @@ export const RewardsDetail: React.FC<IRewardsDetailProps> = ({
...
                   {showApproximateSign && '≈ '}
-                  {weiToNumberFormat(totalEarnedAmount, 6)}
-                  <span className="tw-mr-1">...</span> {asset}
+                  <AssetValue
+                    value={Number(totalEarnedAmount)}
+                    minDecimals={6}
+                    maxDecimals={6}
+                    assetString={asset}
+                    mode={AssetValueMode.auto}
+                  />
```
Comment (tiltom) [inline src/app/pages/RewardPage/components/RewardsDetail/index.tsx:123]:
```suggestion
                  <AssetValue
                    value={Number(totalEarnedAmount)}
                    minDecimals={6}
                    maxDecimals={6}
                    assetString={asset}
                    mode={AssetValueMode.auto}
                    isApproximation={showApproximateSign}
                  />
```

File: src/app/pages/RewardPage/components/RewardsDetail/index.tsx [hunk 6]
```diff
@@ -106,11 +111,16 @@ export const RewardsDetail: React.FC<IRewardsDetailProps> = ({
         <LoadableValue
           value={
             bignumber(totalEarnedAmount).greaterThan(0) ? (
-              <Tooltip content={`${weiTo18(totalEarnedAmount)} ${asset}`}>
+              <Tooltip content={`${totalEarnedAmount} ${asset}`}>
                 <>
                   {showApproximateSign && '≈ '}
```
Comment (tiltom) [inline src/app/pages/RewardPage/components/RewardsDetail/index.tsx:116]:
```suggestion

```

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by tiltom — APPROVED** [review]

File: src/app/pages/RewardPage/components/RewardsDetail/index.tsx [hunk 7]
```diff
@@ -84,11 +84,16 @@ export const RewardsDetail: React.FC<IRewardsDetailProps> = ({
...
             value={
               bignumber(availableAmount).greaterThan(0) ? (
-                <Tooltip content={`${weiTo18(availableAmount)} ${asset}`}>
+                <Tooltip content={`${availableAmount} ${asset}`}>
                   <>
-                    {showApproximateSign && '≈ '}
-                    {weiToNumberFormat(availableAmount, 6)}
-                    <span className="tw-mr-1">...</span> {asset}
+                    <AssetValue
+                      value={Number(availableAmount)}
```
Comment (soulBit) [inline src/app/pages/RewardPage/components/RewardsDetail/index.tsx:90]: Due to the `weiToXX()` formatting calls being omitted here now, the fees totals shown on Fees Earned tab are broken - please update `src\app\pages\RewardPage\components\FeesEarnedTab\index.tsx` with the following change (addition of `weiTo18()`'s):

```
 <RewardsDetail
          color={RewardsDetailColor.Yellow}
          title={t(translations.rewardPage.fee.stakingFee)}
          availableAmount={weiTo18(amountToClaim)}
          totalEarnedAmount={weiTo18(totalRewardsEarned)}
          asset={Asset.RBTC}
          loading={totalAmountLoading}
          showApproximateSign
        />
```

**Review by soulBit — CHANGES_REQUESTED** [review]

**Pull request merged**